### DISCUSSION TITLE
google workspace drive custom

### DISCUSSION THREAD

**Discussion by ravikumarappala** [langflow-ai/langflow#4393] [General]
I am exploring the Google Drive component to see if I can set up Google Drive integration initially and then use a file picker to choose documents when needed. I haven't been able to find information on how the Google Drive component is set up. My goal is to customize it so that I only need to set up the Drive connection once and can select files as required afterward. Any suggestions on how to achieve this customization would be greatly appreciated!

**Comment by RoshansaiK** [comment]
To integrate Google Drive with your application, you can use the Google Drive API to authenticate once and store the credentials. Afterward, use the file picker API to select documents when needed. Look into the Google API client libraries for your chosen language to simplify this process. Be sure to manage OAuth tokens to avoid repeated authentications.

**Comment by ravikumarappala** [comment]
I have done google authentication using Google service Account, but adding
a user based permissions before showing the user folders is not working. As
this is an organizational google workspace, the user has already logged
into the application before, the file picker needs to show files per
permissions.
I am having a bit of a challenge getting this working based on permissions
without having another user authentication as SA is already set up.

Any help is greatly appreciated.
